src/main/java/core/basesyntax/FruitTransaction.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax;
+
+public class FruitTransaction {
+    private Operation operation;
+    private String fruit;
+    private int quantity;
+
+    public FruitTransaction(Operation operation, String fruit, int quantity) {
+        this.operation = operation;
+        this.fruit = fruit;
+        this.quantity = quantity;
+    }
+
+    public Operation getOperation() {
+        return operation;
+    }
+
+    public void setOperation(Operation operation) {
+        this.operation = operation;
+    }
+
+    public String getFruit() {
+        return fruit;
+    }
+
+    public void setFruit(String fruit) {
+        this.fruit = fruit;
+    }
+
+    public int getQuantity() {
+        return quantity;
+    }
+
+    public void setQuantity(int quantity) {
+        this.quantity = quantity;
+    }
+
+    public enum Operation {
+        BALANCE,
+        SUPPLY,
+        PURCHASE,
+        RETURN
+    }
```

Comment:
The `Operation` enum is declared inside the `FruitTransaction` class. It's a good practice to place enums in their own files, especially if they might be used by other classes in the future. Consider moving `Operation` to a separate file.